VS_Meadow_Extension/VS_Meadow_Extension.Shared/MeadowDeployProvider.cs

```diff
@@ -48,82 +50,87 @@ public MeadowDeployProvider(ConfiguredProject configuredProject)
 
-            try
-            {
-                await connection.WaitForMeadowAttach();
+                connection.FileWriteProgress += MeadowConnection_DeploymentProgress;
+                connection.DeviceMessageReceived += MeadowConnection_DeviceMessageReceived;
 
-                if (await connection.IsRuntimeEnabled() == true)
+                try
                 {
-                    await connection.RuntimeDisable();
-                }
+                    await connection.WaitForMeadowAttach();
 
-                var deviceInfo = await connection.GetDeviceInfo();
+                    if (await connection.IsRuntimeEnabled() == true)
+                    {
+                        await connection.RuntimeDisable();
+                    }
 
-                string osVersion = deviceInfo.OsVersion;
+                    var deviceInfo = await connection.GetDeviceInfo();
 
-                var fileManager = new FileManager(null);
-                await fileManager.Refresh();
+                    string osVersion = deviceInfo.OsVersion;
 
-                bool includePdbs = configuredProject?.ProjectConfiguration?.Dimensions["Configuration"].Contains("Debug") ?? false;
+                    var fileManager = new FileManager(null);
+                    await fileManager.Refresh();
 
-                var packageManager = new PackageManager(fileManager);
+                    bool includePdbs = configuredProject?.ProjectConfiguration?.Dimensions["Configuration"].Contains("Debug") ?? false;
 
-                outputLogger.Log("Trimming application binaries...");
+                    var packageManager = new PackageManager(fileManager);
 
-                await packageManager.TrimApplication(new FileInfo(Path.Combine(outputPath, "App.dll")), osVersion, includePdbs, cancellationToken: cancellationToken);
+                    outputLogger.Log("Trimming application binaries...");
+                    await packageManager.TrimApplication(new FileInfo(Path.Combine(outputPath, "App.dll")), osVersion, includePdbs, cancellationToken: cancellationToken);
 
-                await Task.Run(async () => await AppManager.DeployApplication(packageManager, connection, osVersion, outputPath, includePdbs, false, outputLogger, cancellationToken));
+                    outputLogger.Log("Deploying application...");
+                    await AppManager.DeployApplication(packageManager, connection, osVersion, outputPath, includePdbs, false, outputLogger, cancellationToken);
 
-                await connection.RuntimeEnable();
+                    await connection.RuntimeEnable();
 
-                await outputLogger.ShowBuildOutputPane();
-            }
-            finally
-            {
-            }
+                    await outputLogger.ShowBuildOutputPane();
+                }
+                finally
+                {
+                    connection.FileWriteProgress -= MeadowConnection_DeploymentProgress;
```

Comment:
I think I removed this because it fixed the behavior on consecutive deploys but let's leave it in ... we can always remove it later if needed.